.github/workflows/deploy_sdist.yaml

```diff
@@ -0,0 +1,27 @@
+name: Deploy sdist
+
+on:
+  release:
+    types:
+      - published
```

Comment:
Do you want to enable this for pull requests too? The upload to PyPI looks like it is limited to release publishes so might be OK? Oh I guess pushes to `main` would be part of this too.